apps/api/validators.py

```diff
@@ -1,12 +1,384 @@
-from django.core.exceptions import ValidationError
+'''Validates if the file given is a csv file'''
 
 import csv
+from re import match
+from django.core.exceptions import ValidationError
+
+
+class ColumnError(Exception):
+    '''Exception raised if the number of columns in the csv file header does
+    not match the rest'''
+    pass
+
+
+class MultipleUtoridOccurences(Exception):
+    '''Exception raised if the same utor id occurs multiple times in a file
+    '''
+    pass
+
+
+class InvalidUtorid(Exception):
+    '''Utor id not valid. It has to be 8 characters long and only contain
+    alphanumberic symbols
+    '''
+    pass
+
+
+class InvalidData(Exception):
```

Comment:
Similar to ColumnError. Varying data name and row number. Sometimes column number is passed in.

Possible Solution:

```
class InvalidData(Exception):
	def __init__(self, data_name, row, col=None):
		msg = ‘Invalid {} at row: {}’.format(data_name, row)
		if col:
			msg += ‘, column: {}.format(col)
		super(InvalidData, self).__init__(msg)
```
Ex. raise InvalidData(“last name”, 4)
Invalid last name at row: 4

Ex. raise InvalidData(“utorid”, 8, 3)
Invalid utorid at row: 8, column: 3